eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -52,15 +52,17 @@ import scala.concurrent.{Await, ExecutionContext, Future, Promise}
   *
   * @param datadir  directory where eclair-core will write/read its data
   * @param overrideDefaults
-  * @param actorSystem
   * @param seed_opt optional seed, if set eclair will use it instead of generating one and won't create a seed.dat file.
   */
-class Setup(datadir: File, overrideDefaults: Config = ConfigFactory.empty(), actorSystem: ActorSystem = ActorSystem(), seed_opt: Option[BinaryData] = None) extends Logging {
+class Setup(datadir: File,
+            overrideDefaults: Config = ConfigFactory.empty(),
+            seed_opt: Option[BinaryData] = None)(implicit system: ActorSystem) extends Logging {
```

Comment:
Not sure why we did that differently to be honest. Maybe keep the default value though?